stdlib/public/core/Runtime.swift.gyb

```diff
@@ -350,11 +350,24 @@ func _float${bits}ToStringImpl(
 
 func _float${bits}ToString(_ value: Float${bits}, debug: Bool) -> String {
 
-  if value.isInfinite {
-    return value.sign == .minus ? "-inf" : "inf"
-  }
-  if value.isNaN {
-    return "nan"
+  if !value.isFinite {
+    let significand = value.significandBitPattern
+    if significand == 0 {
+      // Infinity
+      return value.sign == .minus ? "-inf" : "inf"
+    }
+    else {
+      // NaN
+      if !debug {
+        return "nan"
+      }
+      let isSignaling = (significand & Float${bits}._quietNaNMask) == 0
+      let payload = significand & ((Float${bits}._quietNaNMask >> 1) - 1)
+      return
```

Comment:
Please add a `FIXME` note here, as these string manipulations are inefficient enough to make someone want to optimize them.